src/include/duckdb/storage/storage_manager.hpp

```diff
@@ -16,19 +16,49 @@
 namespace duckdb {
 class BlockManager;
 class Catalog;
+class CheckpointWriter;
 class DatabaseInstance;
 class TransactionManager;
 class TableCatalogEntry;
 
+struct DatabaseSize {
+	idx_t total_blocks{0};
+	idx_t block_size{0};
+	idx_t free_blocks{0};
+	idx_t used_blocks{0};
+	idx_t bytes{0};
+	idx_t wal_size{0};
+};
+
+class TableIoManager {
+public:
+	// The following would make more sense, but not all indexes are named.
+	// It's not clear how to identify one index vs. another (for constraints).
+	//    virtual BlockManager &GetBlockManagerForIndex(string index_name) = 0;
+	// So do this instead:
+	virtual BlockManager &GetIndexBlockManager() = 0;
+
+	// Get block manager for a new row group.
+	virtual BlockManager &GetBlockManagerForRowData() = 0;
```

Comment:
It could. The DataTable already has a reference to the TableIoManager, which I think it needs regardless. A reference to the BlockManager in the DataTableInfo wouldn't obviate that, but it would let you avoid making this call for each RG.